src/content/en/fundamentals/native-hardware/capturing-images/index.md

```diff
@@ -2,210 +2,343 @@ project_path: /web/_project.yaml
-
 ### Acquire access to the camera
 
-We can directly access a camera and microphone by using an API in the WebRTC 
-specification called `getUserMedia()`. This will prompt the user for 
+We can directly access a camera and microphone by using an API in the WebRTC
+specification called `getUserMedia()`. This will prompt the user for
 access to their connected microphones and cameras.
 
+Support for `getUserMedia` is pretty good, but it isn't yet everywhere. In particular, it is not
+available in Safari 10 or lower, which at the time of writing is still the latest stable version.
+However, [Apple have announced](https://webkit.org/blog/7726/announcing-webrtc-and-media-capture/)
+that it will be available in Safari 11.
+
+It's very simple to detect support, however.
+
+    const supported = 'mediaDevices' in navigator;
+
+Warning: Direct access to the camera is a powerful feature. It requires consent
+from the user, and your site MUST be on a secure origin (HTTPS).
+
+When you call `getUserMedia`, you need to pass in an object that describes what kind of media you
+want. These choices are called constraints.
+
+There are a several possible constraints, covering things like whether you prefer a front- or
+rear-facing camera, whether you want audio, and your preferred resolution for the stream.
+
+To get data from the camera, however, we need just one constraint, and that is `video: true`.
+
 If successful the API will return a `MediaStream` that contains data from
 the camera, and we can then either attach it to a `<video>` element and play it
 to show a real time preview, or attach it to a `<canvas>` to get a
 snapshot.
 
-To get data from the camera we just set `video: true` in the constraints 
-object that is passed to the `getUserMedia()` API.
-
     <video id="player" controls autoplay></video>
-    <script>  
-      var player = document.getElementById('player');
+    <script>
+      const player = document.getElementById('player');
 
-      var handleSuccess = function(stream) {
-        player.srcObject = stream;
+      const constraints = {
+        video: true,
       };
 
-      navigator.mediaDevices.getUserMedia({video: true})
-          .then(handleSuccess);
+      navigator.mediaDevices.getUserMedia(constraints)
+        .then((stream) => {
+          player.srcObject = stream;
+        });
     </script>
 
 By itself, this isn't that useful. All we can do is take the video data and play
-it back.
+it back. If we want to get an image, we have to do a little extra work.
 
-### Take a snapshot from the camera
+### Grab a snapshot
+Our best supported option for getting an image is to draw a frame from the video to a canvas.
 
-To access the raw data from the camera we have to take the stream created by
-`getUserMedia()` and process the data. Unlike `Web Audio`, there isn't a 
-dedicated stream processing API for video on the web so we have to resort to 
+Unlike `Web Audio`, there isn't a
```

Comment:
Change to 'Unlike the Web Audio API'. Notice that I didn't use back ticks. If there were such an object `WebAudio` would be another option.